generic-ec-core/src/lib.rs

```diff
@@ -248,3 +248,12 @@ pub trait Reduce<const N: usize> {
     /// integer modulo curve (prime) order
     fn from_le_array_mod_order(bytes: &[u8; N]) -> Self;
 }
+
+/// Marker trait for curves whose underlying implementation doesn't allow
+/// representing invalid points.
+/// # Safety
+/// Safe to implement when the checks in `generic_ec::TryFromRaw` would always
+/// return 1 for any point. Those checks are:
+/// - `point.is_on_curve()`
+/// - `point.is_torsion_free()`
+pub unsafe trait NoInvalidPoints {}
```

Comment:
It is strange to refer to the `generic_ec::TryFromRaw` trait which is located in upstream library. Maybe `NoInvalidPoints` should be in `generic_ec`?

Reply:
If it's in generic-ec, then I can't refer to it to impl it in generic-ec-curves, right?

Reply:
You can, as it is a trait first defined in generic_ec

Reply:
If you define a trait in your crate, you can implement it on whatever structs you want. Other crates, however, will only be able to implement your trait only on the types defined in those crates.

Reply:
You mean I would impl it in generic-ec as well? I thought for an unsafe trait it's better to keep it closer to the curves definitions, and also to have less of `#[cfg(feature)]` pollution

Reply:
Yes, curves crate can't possibly implement trait from `generic-ec` as it's upstream, it would cause circular dependency.

> I thought for an unsafe trait it's better to keep it closer to the curves definitions

I think, generally it's a good practice, but I also think it's acceptable to do otherwise in this case

Reply:
> and also to have less of `#[cfg(feature)]` pollution

That's just one `#[cfg(feature)]` per each impl

Reply:
I mean, you can keep it in the core if you prefer, I just suggest to rewrite/update the comment to not refer to a train in upstream crate

Reply:
I agree with you, it is a bit silly to ask the implementers to look at the downstream crate. I'll think about it a bit